assets/app/scripts/controllers/create/createFromImage.js

```diff
@@ -0,0 +1,223 @@
+"use strict";
+
+angular.module("openshiftConsole")
+  .factory("CreateFromImageControllerMixin", function(){
+    
+    function sanatize(name){
+      return name.toLowerCase();
+    };
+    
+    return {
+      
+      /**
+       * Update the routing uri based on
+       * a users UI choice
+       */
+      updateURI: function(){
+        if(this.name === undefined){
+          this.routing.uri = "";
+        }
+        else
+        {
+          if(this.routing.option !== 'custom'){
+            this.routing.uri =  this.name + ".openshift.com";
+          }
+          this.routing.uri = sanatize(this.routing.uri);
+        }
+      }
+    };
+  })
+  .controller("CreateFromImageController", function ($scope,
+      Logger,
+      $q,
+      $routeParams, 
+      DataService, 
+      Navigate, 
+      NameGenerator, 
+      ApplicationGenerator,
+      TaskList,
+      CreateFromImageControllerMixin
+    ){
+    angular.extend($scope, CreateFromImageControllerMixin);
+    function initAndValidate(scope){
+
+      if(!$routeParams.imageName){
+        Navigate.toErrorPage("Cannot create from source: a base image was not specified");
+      }
+      if(!$routeParams.imageTag){
+        Navigate.toErrorPage("Cannot create from source: a base image tag was not specified");
+      }
+      if(!$routeParams.sourceURL){
+        Navigate.toErrorPage("Cannot create from source: source url was not specified");
+      }
+
+      scope.emptyMessage = "Loading...";
+      scope.imageName = $routeParams.imageName;
+      scope.imageTag = $routeParams.imageTag;
+      scope.namespace = $routeParams.namespace;
+      scope.buildConfig = {
+        sourceUrl: $routeParams.sourceURL,
+        buildOnSourceChange: true,
+        buildOnImageChange: true
+      };
+      scope.deploymentConfig = {
+        deployOnNewImage: true,
+        deployOnConfigChange: true,
+        envVars : {
+        }
+      };
+      scope.routing = {
+        option: 'none',
+        protocol: 'http,https',
+        uri: "",
+        customCerts: false,
+        certificate: "",
+        key: "",
+        caCertificate: ""
+      };
+      scope.labels = {};
+      scope.scaling = {
+        replicas: 1
+      };
+
+      DataService.get("imageRepositories", scope.imageName, scope, {namespace: scope.namespace})
```

Comment:
do you need to retrieve both the imageRepo and the imageRepoTags ? 

Also you should switch these to use imageStreams and imageStreamTags (would need to rebase on master from friday)